src/app/common/hooks/use-convert-to-fiat-amount.ts

```diff
@@ -2,29 +2,64 @@ import { useCallback, useMemo } from 'react';
 
 import { CryptoCurrencies } from '@shared/models/currencies.model';
 import { createMarketData, createMarketPair } from '@shared/models/market.model';
-import type { Money } from '@shared/models/money.model';
+import { type Money, createMoney } from '@shared/models/money.model';
+import { isUndefined } from '@shared/utils';
 
-import { useCryptoCurrencyMarketData } from '@app/query/common/market-data/market-data.hooks';
+import { i18nFormatCurrency } from '@app/common/money/format-money';
+import { unitToFractionalUnit } from '@app/common/money/unit-conversion';
+import { useCryptoCurrencyMarketDataMean } from '@app/query/common/market-data/market-data.hooks';
 
 import { baseCurrencyAmountInQuote } from '../money/calculate-money';
 
-export function useConvertCryptoCurrencyToFiatAmount(currency: CryptoCurrencies) {
-  const cryptoCurrencyMarketData = useCryptoCurrencyMarketData(currency);
+export function useConvertCryptoCurrencyToFiatAmountMean(currency: CryptoCurrencies) {
+  const cryptoCurrencyMarketData = useCryptoCurrencyMarketDataMean(currency);
 
   return useCallback(
     (value: Money) => baseCurrencyAmountInQuote(value, cryptoCurrencyMarketData),
     [cryptoCurrencyMarketData]
   );
 }
 
-export function useConvertAlexSdkCurrencyToFiatAmount(currency: CryptoCurrencies, price: Money) {
-  const alexCurrencyMarketData = useMemo(
+function useConvertCryptoCurrencyToFiatAmount(currency: CryptoCurrencies, price: Money) {
+  const cryptoCurrencyMarketData = useMemo(
     () => createMarketData(createMarketPair(currency, 'USD'), price),
     [currency, price]
   );
 
   return useCallback(
-    (value: Money) => baseCurrencyAmountInQuote(value, alexCurrencyMarketData),
-    [alexCurrencyMarketData]
+    (value: Money) => baseCurrencyAmountInQuote(value, cryptoCurrencyMarketData),
+    [cryptoCurrencyMarketData]
+  );
+}
+
+export function useFormattedFractionalUnitAsFiat(balance?: Money, price?: Money, value?: string) {
+  const convertAlexSdkCurrencyToUsd = useConvertCryptoCurrencyToFiatAmount(
+    balance?.symbol ?? '',
+    price ?? createMoney(0, 'USD')
+  );
+
+  if (isUndefined(balance) || isUndefined(price) || isUndefined(value)) return;
+
+  const convertedAmountAsMoney = convertAlexSdkCurrencyToUsd(
+    createMoney(unitToFractionalUnit(balance.decimals)(value), balance.symbol, balance.decimals)
   );
+
+  if (convertedAmountAsMoney.amount.isNaN()) return;
+  return i18nFormatCurrency(convertedAmountAsMoney);
```

Comment:
IMO formatting should be done at the component/jsx level. What if I want to programatically read/modify this value? Not possible because of decision to pre-format value.

Reply:
Ok, I can move the formatting and rethink these.